src/lightning_lite/plugins/precision/deepspeed.py

```diff
@@ -53,18 +56,23 @@ def __init__(self, precision: Union[str, int], amp_type: str, amp_level: Optiona
 
             amp_level = amp_level or "O2"
 
-        supported_precision = (PrecisionType.HALF, PrecisionType.FLOAT, PrecisionType.BFLOAT)
-        if precision not in supported_precision:
```

Comment:
This was a bug, right? Comparing the precision (int) with the enum strings would never work here, i.e., the if-condition would be false.

Reply:
No. Our enums are magic :sparkles: and convert to str in the equality check: https://github.com/Lightning-AI/utilities/blob/86e9e48e453ae9690448a0d3cef320d3dc4bd777/src/lightning_utilities/core/enums.py#L23

Reply:
That explains it .Thanks